NeuroPilot/ContextPatches.cs

```diff
@@ -0,0 +1,256 @@
+            sendContext("Ship Log Context", $"[NEW SHIP LOG FACT] {astroObjectName} - {shipLogEntry.GetName(false)}: {__instance.GetText()}");
+        }
+
+        [HarmonyPostfix, HarmonyPatch(typeof(SignalscopeUI), nameof(SignalscopeUI.UpdateLabels))]
+        public static void SignalscopeUI_UpdateLabels_Postfix(SignalscopeUI __instance)
+        {
+            AudioSignal strongestSignal = __instance._signalscopeTool.GetStrongestSignal();
+            if (_lastSignal == strongestSignal && Time.time - _lastSignalTime < 10f) return;
+            if (strongestSignal == null || strongestSignal.GetSignalStrength() < 0.9f) return;
+            string text = PlayerData.KnowsSignal(strongestSignal.GetName()) ? AudioSignal.SignalNameToString(strongestSignal.GetName()) : UITextLibrary.GetString(UITextType.UnknownSignal);
+            sendContext("Signalscope Context", $"Listening to Signal: '{text}' | Distance: {Mathf.Round(strongestSignal.GetDistanceFromScope())}m | Frequency: '{AudioSignal.FrequencyToString(strongestSignal._frequency, false)}'");
+            _lastSignalTime = Time.time;
+            _lastSignal = strongestSignal;
+        }
+
+        [HarmonyPostfix, HarmonyPatch(typeof(GlobalMessenger), nameof(GlobalMessenger.FireEvent))]
+        public static void GlobalMessenger_FireEvent_Postfix(string eventType)
+        {
+            switch (eventType)
+            {
+                case "PlayerEnterQuantumMoon":
+                    sendContext("Location Context", "Player entered the Quantum Moon");
+                    break;
+                case "PlayerExitQuantumMoon":
+                    sendContext("Location Context", "Player left the Quantum Moon");
+                    break;
+                case "PlayerEnterBlackHole":
+                    sendContext("Location Context", "Player entered a Black Hole");
+                    break;
+                case "PlayerEscapedTimeLoop":
+                    sendContext("Death Context", "Player escaped the time loop");
+                    break;
+            }
+        }
+
+        [HarmonyPostfix, HarmonyPatch(typeof(Campfire), nameof(Campfire.StartRoasting))]
+        public static void Campfire_StartRoasting_Postfix()
+        {
+            sendContext("Misc Context", $"Player started roasting a marshmallow");
+        }
+
+        [HarmonyPrefix, HarmonyPatch(typeof(Marshmallow), nameof(Marshmallow.Eat))]
+        public static void Marshmallow_Eat_Prefix(Marshmallow __instance)
+        {
+            sendContext("Misc Context", $"Player ate a {(__instance.IsBurned() ? "burned " : "")}marshmallow");
+        }
+
+        [HarmonyPostfix, HarmonyPatch(typeof(Campfire), nameof(Campfire.StartSleeping))]
+        public static void Campfire_StartSleeping_Postfix()
+        {
+            sendContext("Misc Context", $"Player started sleeping at a campfire");
+        }
+
+        private static string getTextFromDialogue(CharacterDialogueTree characterDialogueTree)
```

Comment:
Standard .NET naming keeps methods capitalized even if not public. I might go in and rename these afterwards but not a blocker